mods/persistence/modules/recycling/recycler.dm

```
@@ -0,0 +1,383 @@
+		/obj/item/stock_parts/keyboard = 1,
+		/obj/item/stock_parts/power/apc/buildable = 1,
+		/obj/item/stock_parts/radio/transmitter/basic/buildable,
+		/obj/item/stock_parts/radio/receiver/buildable,
+	)
+
+///////////////////////
+// Recycler
+///////////////////////
+/obj/machinery/recycler
+	name = "recycler"
+	desc = "Process trash into raw materials."
+	icon = 'icons/obj/recycling.dmi'
+	icon_state = "grinder-o0"
+	layer = ABOVE_HUMAN_LAYER
+	density = TRUE
+	anchored = TRUE
+	use_power = POWER_USE_IDLE
+	idle_power_usage = 1 KILOWATTS		//This is vastly lower than it should for gameplay reasons. A 200hp engine running at 25% power would need 37 kW
+	active_power_usage = 40 KILOWATTS	//This is vastly lower than it should for gameplay reasons. 200 HP electric motor at 100% efficiency would need 149 kW
+	required_interaction_dexterity = DEXTERITY_SIMPLE_MACHINES
+
+	construct_state = /decl/machine_construction/default/panel_closed
+	uncreated_component_parts = list(
+		/obj/item/stock_parts/power/apc/buildable,
+		/obj/item/stock_parts/radio/transmitter/basic/buildable,
+		/obj/item/stock_parts/radio/receiver/buildable,
+	)
+	stock_part_presets = list(
+		/decl/stock_part_preset/radio/basic_transmitter/recycler = 1, 
+		/decl/stock_part_preset/radio/receiver/recycler = 1,
+	)
+	public_variables = list(
+		/decl/public_access/public_variable/identifier,
+		/decl/public_access/public_variable/use_power,
+		/decl/public_access/public_variable/reagents,
+		/decl/public_access/public_variable/reagents/volumes,
+		/decl/public_access/public_variable/reagents/free_space,
+		/decl/public_access/public_variable/reagents/total_volume,
+		/decl/public_access/public_variable/reagents/maximum_volume,
+	)
+	public_methods = list(
+		/decl/public_access/public_method/toggle_power,
+	)
+
+	var/bloodied 				= FALSE		//Whether we show the sprite with blood or not
+	var/tmp/grinding			= FALSE 	//Whether we're currently busy grinding something down
```

Why the tmp variables? While I think there might be some very old usages of issaved in our saving system, since we're working off a whitelist these really shouldn't be necessary, unless there's something I'm missing.

---

Its good practice to mark runtime vars tmp, and constants const. Also the code still checks issaved, on vars even in the white list.

---

Because /tmp/ isn't used elsewhere, I disagree that this is good practice.  It only confuses things and will make saving the var fail silently if someone does add it down the line (even erroneously) . I think really that issaved() check should be removed.

I'd be more willing to formalize use of /tmp/ if we could modify upstream variables to use it. Otherwise it's just a confusing standard that could only ever be applied in the persistence modpack.

---

Just look at the nebula code. A lot of new code they made respects proper /tmp use. 

And I'm gonna be blunt here, I'm not sure you really know what you're talking about here? We don't wanna remove is_saved. A lot of variables in base byond code aren't actual variables, and if some clown decides to add them to the saved list without that check we could easily break things and spend hours trying to figure out why... This is a particularity of the language we're using, we can't just ignore it and pretend everything is fine.

The solution to fix silent failing is to actually make checks return errors, which most of the checks in the current saving code never do right now for some reasons. To remove the actual checks instead is nonsensical.

---

Using /tmp/ for runtime vars is fine. I wasn't aware of this use upstream. To be honest, I'm still a little tepid on using the built in issaved() proc but it's probably not bad practice. The silent erroring should be fixed at some point, but outside of this PR.